<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="utf-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <meta name="generator" content="rustdoc">
    <meta name="description" content="API documentation for the Rust `Read` struct in crate `google_genomics1`.">
    <meta name="keywords" content="rust, rustlang, rust-lang, Read">

    <title>google_genomics1::Read - Rust</title>

    <link rel="stylesheet" type="text/css" href="../rustdoc.css">
    <link rel="stylesheet" type="text/css" href="../main.css">
    

    
    
</head>
<body class="rustdoc">
    <!--[if lte IE 8]>
    <div class="warning">
        This old browser is unsupported and will most likely display funky
        things.
    </div>
    <![endif]-->

    

    <nav class="sidebar">
        
        <p class='location'><a href='index.html'>google_genomics1</a></p><script>window.sidebarCurrent = {name: 'Read', ty: 'struct', relpath: ''};</script><script defer src="sidebar-items.js"></script>
    </nav>

    <nav class="sub">
        <form class="search-form js-only">
            <div class="search-container">
                <input class="search-input" name="search"
                       autocomplete="off"
                       placeholder="Click or press ‘S’ to search, ‘?’ for more options…"
                       type="search">
            </div>
        </form>
    </nav>

    <section id='main' class="content struct">
<h1 class='fqn'><span class='in-band'>Struct <a href='index.html'>google_genomics1</a>::<wbr><a class='struct' href=''>Read</a></span><span class='out-of-band'><span id='render-detail'>
                   <a id="toggle-all-docs" href="javascript:void(0)" title="collapse all docs">
                       [<span class='inner'>&#x2212;</span>]
                   </a>
               </span><a id='src-7808' class='srclink' href='../src/google_genomics1/Users/byron/dev/rs/google-apis-rs/target/debug/build/google-genomics1-1ad8d3924bbaea8d/out/lib.rs.html#17293-17328' title='goto source code'>[src]</a></span></h1>
<pre class='rust struct'>pub struct Read {
    pub read_group_set_id: <a class='enum' href='https://doc.rust-lang.org/nightly/core/option/enum.Option.html' title='core::option::Option'>Option</a>&lt;<a class='struct' href='https://doc.rust-lang.org/nightly/collections/string/struct.String.html' title='collections::string::String'>String</a>&gt;,
    pub failed_vendor_quality_checks: <a class='enum' href='https://doc.rust-lang.org/nightly/core/option/enum.Option.html' title='core::option::Option'>Option</a>&lt;<a class='primitive' href='https://doc.rust-lang.org/nightly/std/primitive.bool.html'>bool</a>&gt;,
    pub fragment_name: <a class='enum' href='https://doc.rust-lang.org/nightly/core/option/enum.Option.html' title='core::option::Option'>Option</a>&lt;<a class='struct' href='https://doc.rust-lang.org/nightly/collections/string/struct.String.html' title='collections::string::String'>String</a>&gt;,
    pub next_mate_position: <a class='enum' href='https://doc.rust-lang.org/nightly/core/option/enum.Option.html' title='core::option::Option'>Option</a>&lt;<a class='struct' href='../google_genomics1/struct.Position.html' title='google_genomics1::Position'>Position</a>&gt;,
    pub supplementary_alignment: <a class='enum' href='https://doc.rust-lang.org/nightly/core/option/enum.Option.html' title='core::option::Option'>Option</a>&lt;<a class='primitive' href='https://doc.rust-lang.org/nightly/std/primitive.bool.html'>bool</a>&gt;,
    pub fragment_length: <a class='enum' href='https://doc.rust-lang.org/nightly/core/option/enum.Option.html' title='core::option::Option'>Option</a>&lt;<a class='primitive' href='https://doc.rust-lang.org/nightly/std/primitive.i32.html'>i32</a>&gt;,
    pub id: <a class='enum' href='https://doc.rust-lang.org/nightly/core/option/enum.Option.html' title='core::option::Option'>Option</a>&lt;<a class='struct' href='https://doc.rust-lang.org/nightly/collections/string/struct.String.html' title='collections::string::String'>String</a>&gt;,
    pub alignment: <a class='enum' href='https://doc.rust-lang.org/nightly/core/option/enum.Option.html' title='core::option::Option'>Option</a>&lt;<a class='struct' href='../google_genomics1/struct.LinearAlignment.html' title='google_genomics1::LinearAlignment'>LinearAlignment</a>&gt;,
    pub info: <a class='enum' href='https://doc.rust-lang.org/nightly/core/option/enum.Option.html' title='core::option::Option'>Option</a>&lt;<a class='struct' href='https://doc.rust-lang.org/nightly/std/collections/hash/map/struct.HashMap.html' title='std::collections::hash::map::HashMap'>HashMap</a>&lt;<a class='struct' href='https://doc.rust-lang.org/nightly/collections/string/struct.String.html' title='collections::string::String'>String</a>, <a class='struct' href='https://doc.rust-lang.org/nightly/collections/vec/struct.Vec.html' title='collections::vec::Vec'>Vec</a>&lt;<a class='struct' href='https://doc.rust-lang.org/nightly/collections/string/struct.String.html' title='collections::string::String'>String</a>&gt;&gt;&gt;,
    pub duplicate_fragment: <a class='enum' href='https://doc.rust-lang.org/nightly/core/option/enum.Option.html' title='core::option::Option'>Option</a>&lt;<a class='primitive' href='https://doc.rust-lang.org/nightly/std/primitive.bool.html'>bool</a>&gt;,
    pub number_reads: <a class='enum' href='https://doc.rust-lang.org/nightly/core/option/enum.Option.html' title='core::option::Option'>Option</a>&lt;<a class='primitive' href='https://doc.rust-lang.org/nightly/std/primitive.i32.html'>i32</a>&gt;,
    pub read_number: <a class='enum' href='https://doc.rust-lang.org/nightly/core/option/enum.Option.html' title='core::option::Option'>Option</a>&lt;<a class='primitive' href='https://doc.rust-lang.org/nightly/std/primitive.i32.html'>i32</a>&gt;,
    pub proper_placement: <a class='enum' href='https://doc.rust-lang.org/nightly/core/option/enum.Option.html' title='core::option::Option'>Option</a>&lt;<a class='primitive' href='https://doc.rust-lang.org/nightly/std/primitive.bool.html'>bool</a>&gt;,
    pub read_group_id: <a class='enum' href='https://doc.rust-lang.org/nightly/core/option/enum.Option.html' title='core::option::Option'>Option</a>&lt;<a class='struct' href='https://doc.rust-lang.org/nightly/collections/string/struct.String.html' title='collections::string::String'>String</a>&gt;,
    pub aligned_quality: <a class='enum' href='https://doc.rust-lang.org/nightly/core/option/enum.Option.html' title='core::option::Option'>Option</a>&lt;<a class='struct' href='https://doc.rust-lang.org/nightly/collections/vec/struct.Vec.html' title='collections::vec::Vec'>Vec</a>&lt;<a class='primitive' href='https://doc.rust-lang.org/nightly/std/primitive.i32.html'>i32</a>&gt;&gt;,
    pub aligned_sequence: <a class='enum' href='https://doc.rust-lang.org/nightly/core/option/enum.Option.html' title='core::option::Option'>Option</a>&lt;<a class='struct' href='https://doc.rust-lang.org/nightly/collections/string/struct.String.html' title='collections::string::String'>String</a>&gt;,
    pub secondary_alignment: <a class='enum' href='https://doc.rust-lang.org/nightly/core/option/enum.Option.html' title='core::option::Option'>Option</a>&lt;<a class='primitive' href='https://doc.rust-lang.org/nightly/std/primitive.bool.html'>bool</a>&gt;,
}</pre><div class='docblock'><p>A read alignment describes a linear alignment of a string of DNA to a reference sequence, in addition to metadata about the fragment (the molecule of DNA sequenced) and the read (the bases which were read by the sequencer). A read is equivalent to a line in a SAM file. A read belongs to exactly one read group and exactly one read group set. For more genomics resource definitions, see <a href="https://cloud.google.com/genomics/fundamentals-of-google-genomics">Fundamentals of Google Genomics</a> ### Reverse-stranded reads Mapped reads (reads having a non-null <code>alignment</code>) can be aligned to either the forward or the reverse strand of their associated reference. Strandedness of a mapped read is encoded by <code>alignment.position.reverseStrand</code>. If we consider the reference to be a forward-stranded coordinate space of <code>[0, reference.length)</code> with <code>0</code> as the left-most position and <code>reference.length</code> as the right-most position, reads are always aligned left to right. That is, <code>alignment.position.position</code> always refers to the left-most reference coordinate and <code>alignment.cigar</code> describes the alignment of this read to the reference from left to right. All per-base fields such as <code>alignedSequence</code> and <code>alignedQuality</code> share this same left-to-right orientation; this is true of reads which are aligned to either strand. For reverse-stranded reads, this means that <code>alignedSequence</code> is the reverse complement of the bases that were originally reported by the sequencing machine. ### Generating a reference-aligned sequence string When interacting with mapped reads, it&#39;s often useful to produce a string representing the local alignment of the read to reference. The following pseudocode demonstrates one way of doing this: out = &quot;&quot; offset = 0 for c in read.alignment.cigar { switch c.operation { case &quot;ALIGNMENT_MATCH&quot;, &quot;SEQUENCE_MATCH&quot;, &quot;SEQUENCE_MISMATCH&quot;: out += read.alignedSequence[offset:offset+c.operationLength] offset += c.operationLength break case &quot;CLIP_SOFT&quot;, &quot;INSERT&quot;: offset += c.operationLength break case &quot;PAD&quot;: out += repeat(&quot;*&quot;, c.operationLength) break case &quot;DELETE&quot;: out += repeat(&quot;-&quot;, c.operationLength) break case &quot;SKIP&quot;: out += repeat(&quot; &quot;, c.operationLength) break case &quot;CLIP_HARD&quot;: break } } return out ### Converting to SAM&#39;s CIGAR string The following pseudocode generates a SAM CIGAR string from the <code>cigar</code> field. Note that this is a lossy conversion (<code>cigar.referenceSequence</code> is lost). cigarMap = { &quot;ALIGNMENT_MATCH&quot;: &quot;M&quot;, &quot;INSERT&quot;: &quot;I&quot;, &quot;DELETE&quot;: &quot;D&quot;, &quot;SKIP&quot;: &quot;N&quot;, &quot;CLIP_SOFT&quot;: &quot;S&quot;, &quot;CLIP_HARD&quot;: &quot;H&quot;, &quot;PAD&quot;: &quot;P&quot;, &quot;SEQUENCE_MATCH&quot;: &quot;=&quot;, &quot;SEQUENCE_MISMATCH&quot;: &quot;X&quot;, } cigarStr = &quot;&quot; for c in read.alignment.cigar { cigarStr += c.operationLength + cigarMap[c.operation] } return cigarStr</p>

<h1 id='activities' class='section-header'><a href='#activities'>Activities</a></h1>
<p>This type is used in activities, which are methods you may call on this type or where this type is involved in. 
The list links the activity name, along with information about where it is used (one of <em>request</em> and <em>response</em>).</p>

<ul>
<li><a href="struct.ReadSearchCall.html">search reads</a> (none)</li>
<li><a href="struct.ReadStreamCall.html">stream reads</a> (none)</li>
</ul>
</div><h2 class='fields'>Fields</h2>
<table><tr class='stab '>
                             <td id='structfield.read_group_set_id'><code>read_group_set_id</code></td><td><div class='docblock'><p>The ID of the read group set this read belongs to. A read belongs to exactly one read group set.</p>
</div></td></tr><tr class='stab '>
                             <td id='structfield.failed_vendor_quality_checks'><code>failed_vendor_quality_checks</code></td><td><div class='docblock'><p>Whether this read did not pass filters, such as platform or vendor quality controls (SAM flag 0x200).</p>
</div></td></tr><tr class='stab '>
                             <td id='structfield.fragment_name'><code>fragment_name</code></td><td><div class='docblock'><p>The fragment name. Equivalent to QNAME (query template name) in SAM.</p>
</div></td></tr><tr class='stab '>
                             <td id='structfield.next_mate_position'><code>next_mate_position</code></td><td><div class='docblock'><p>The mapping of the primary alignment of the <code>(readNumber+1)%numberReads</code> read in the fragment. It replaces mate position and mate strand in SAM.</p>
</div></td></tr><tr class='stab '>
                             <td id='structfield.supplementary_alignment'><code>supplementary_alignment</code></td><td><div class='docblock'><p>Whether this alignment is supplementary. Equivalent to SAM flag 0x800. Supplementary alignments are used in the representation of a chimeric alignment. In a chimeric alignment, a read is split into multiple linear alignments that map to different reference contigs. The first linear alignment in the read will be designated as the representative alignment; the remaining linear alignments will be designated as supplementary alignments. These alignments may have different mapping quality scores. In each linear alignment in a chimeric alignment, the read will be hard clipped. The <code>alignedSequence</code> and <code>alignedQuality</code> fields in the alignment record will only represent the bases for its respective linear alignment.</p>
</div></td></tr><tr class='stab '>
                             <td id='structfield.fragment_length'><code>fragment_length</code></td><td><div class='docblock'><p>The observed length of the fragment, equivalent to TLEN in SAM.</p>
</div></td></tr><tr class='stab '>
                             <td id='structfield.id'><code>id</code></td><td><div class='docblock'><p>The server-generated read ID, unique across all reads. This is different from the <code>fragmentName</code>.</p>
</div></td></tr><tr class='stab '>
                             <td id='structfield.alignment'><code>alignment</code></td><td><div class='docblock'><p>The linear alignment for this alignment record. This field is null for unmapped reads.</p>
</div></td></tr><tr class='stab '>
                             <td id='structfield.info'><code>info</code></td><td><div class='docblock'><p>A map of additional read alignment information. This must be of the form map (string key mapping to a list of string values).</p>
</div></td></tr><tr class='stab '>
                             <td id='structfield.duplicate_fragment'><code>duplicate_fragment</code></td><td><div class='docblock'><p>The fragment is a PCR or optical duplicate (SAM flag 0x400).</p>
</div></td></tr><tr class='stab '>
                             <td id='structfield.number_reads'><code>number_reads</code></td><td><div class='docblock'><p>The number of reads in the fragment (extension to SAM flag 0x1).</p>
</div></td></tr><tr class='stab '>
                             <td id='structfield.read_number'><code>read_number</code></td><td><div class='docblock'><p>The read number in sequencing. 0-based and less than numberReads. This field replaces SAM flag 0x40 and 0x80.</p>
</div></td></tr><tr class='stab '>
                             <td id='structfield.proper_placement'><code>proper_placement</code></td><td><div class='docblock'><p>The orientation and the distance between reads from the fragment are consistent with the sequencing protocol (SAM flag 0x2).</p>
</div></td></tr><tr class='stab '>
                             <td id='structfield.read_group_id'><code>read_group_id</code></td><td><div class='docblock'><p>The ID of the read group this read belongs to. A read belongs to exactly one read group. This is a server-generated ID which is distinct from SAM&#39;s RG tag (for that value, see ReadGroup.name).</p>
</div></td></tr><tr class='stab '>
                             <td id='structfield.aligned_quality'><code>aligned_quality</code></td><td><div class='docblock'><p>The quality of the read sequence contained in this alignment record (equivalent to QUAL in SAM). <code>alignedSequence</code> and <code>alignedQuality</code> may be shorter than the full read sequence and quality. This will occur if the alignment is part of a chimeric alignment, or if the read was trimmed. When this occurs, the CIGAR for this read will begin/end with a hard clip operator that will indicate the length of the excised sequence.</p>
</div></td></tr><tr class='stab '>
                             <td id='structfield.aligned_sequence'><code>aligned_sequence</code></td><td><div class='docblock'><p>The bases of the read sequence contained in this alignment record, <strong>without CIGAR operations applied</strong> (equivalent to SEQ in SAM). <code>alignedSequence</code> and <code>alignedQuality</code> may be shorter than the full read sequence and quality. This will occur if the alignment is part of a chimeric alignment, or if the read was trimmed. When this occurs, the CIGAR for this read will begin/end with a hard clip operator that will indicate the length of the excised sequence.</p>
</div></td></tr><tr class='stab '>
                             <td id='structfield.secondary_alignment'><code>secondary_alignment</code></td><td><div class='docblock'><p>Whether this alignment is secondary. Equivalent to SAM flag 0x100. A secondary alignment represents an alternative to the primary alignment for this read. Aligners may return secondary alignments if a read can map ambiguously to multiple coordinates in the genome. By convention, each read has one and only one alignment where both <code>secondaryAlignment</code> and <code>supplementaryAlignment</code> are false.</p>
</div></td></tr></table><h2 id='implementations'>Trait Implementations</h2><h3 class='impl'><span class='in-band'><code>impl <a class='trait' href='../google_genomics1/trait.Resource.html' title='google_genomics1::Resource'>Resource</a> for <a class='struct' href='../google_genomics1/struct.Read.html' title='google_genomics1::Read'>Read</a></code></span><span class='out-of-band'><div class='ghost'></div><a id='src-8251' class='srclink' href='../src/google_genomics1/Users/byron/dev/rs/google-apis-rs/target/debug/build/google-genomics1-1ad8d3924bbaea8d/out/lib.rs.html#18314' title='goto source code'>[src]</a></span></h3>
<div class='impl-items'></div><h3 id='derived_implementations'>Derived Implementations </h3><h3 class='impl'><span class='in-band'><code>impl <a class='trait' href='https://doc.rust-lang.org/nightly/core/default/trait.Default.html' title='core::default::Default'>Default</a> for <a class='struct' href='../google_genomics1/struct.Read.html' title='google_genomics1::Read'>Read</a></code></span><span class='out-of-band'><div class='ghost'></div><a id='src-7826' class='srclink' href='../src/google_genomics1/Users/byron/dev/rs/google-apis-rs/target/debug/build/google-genomics1-1ad8d3924bbaea8d/out/lib.rs.html#17292' title='goto source code'>[src]</a></span></h3>
<div class='impl-items'><h4 id='method.default' class='method'><code>fn <a href='https://doc.rust-lang.org/nightly/core/default/trait.Default.html#tymethod.default' class='fnname'>default</a>() -&gt; <a class='struct' href='../google_genomics1/struct.Read.html' title='google_genomics1::Read'>Read</a></code></h4>
<div class='docblock'><p>Returns the &quot;default value&quot; for a type. <a href="https://doc.rust-lang.org/nightly/core/default/trait.Default.html#tymethod.default">Read more</a></p>
</div></div><h3 class='impl'><span class='in-band'><code>impl <a class='trait' href='https://doc.rust-lang.org/nightly/core/clone/trait.Clone.html' title='core::clone::Clone'>Clone</a> for <a class='struct' href='../google_genomics1/struct.Read.html' title='google_genomics1::Read'>Read</a></code></span><span class='out-of-band'><div class='ghost'></div><a id='src-7828' class='srclink' href='../src/google_genomics1/Users/byron/dev/rs/google-apis-rs/target/debug/build/google-genomics1-1ad8d3924bbaea8d/out/lib.rs.html#17292' title='goto source code'>[src]</a></span></h3>
<div class='impl-items'><h4 id='method.clone' class='method'><code>fn <a href='https://doc.rust-lang.org/nightly/core/clone/trait.Clone.html#tymethod.clone' class='fnname'>clone</a>(&amp;self) -&gt; <a class='struct' href='../google_genomics1/struct.Read.html' title='google_genomics1::Read'>Read</a></code></h4>
<div class='docblock'><p>Returns a copy of the value. <a href="https://doc.rust-lang.org/nightly/core/clone/trait.Clone.html#tymethod.clone">Read more</a></p>
</div><h4 id='method.clone_from' class='method'><code>fn <a href='https://doc.rust-lang.org/nightly/core/clone/trait.Clone.html#method.clone_from' class='fnname'>clone_from</a>(&amp;mut self, source: &amp;Self)</code><div class='since' title='Stable since Rust version 1.0.0'>1.0.0</div></h4>
<div class='docblock'><p>Performs copy-assignment from <code>source</code>. <a href="https://doc.rust-lang.org/nightly/core/clone/trait.Clone.html#method.clone_from">Read more</a></p>
</div></div><h3 class='impl'><span class='in-band'><code>impl <a class='trait' href='https://doc.rust-lang.org/nightly/core/fmt/trait.Debug.html' title='core::fmt::Debug'>Debug</a> for <a class='struct' href='../google_genomics1/struct.Read.html' title='google_genomics1::Read'>Read</a></code></span><span class='out-of-band'><div class='ghost'></div><a id='src-7848' class='srclink' href='../src/google_genomics1/Users/byron/dev/rs/google-apis-rs/target/debug/build/google-genomics1-1ad8d3924bbaea8d/out/lib.rs.html#17292' title='goto source code'>[src]</a></span></h3>
<div class='impl-items'><h4 id='method.fmt' class='method'><code>fn <a href='https://doc.rust-lang.org/nightly/core/fmt/trait.Debug.html#tymethod.fmt' class='fnname'>fmt</a>(&amp;self, __arg_0: &amp;mut <a class='struct' href='https://doc.rust-lang.org/nightly/core/fmt/struct.Formatter.html' title='core::fmt::Formatter'>Formatter</a>) -&gt; <a class='type' href='https://doc.rust-lang.org/nightly/core/fmt/type.Result.html' title='core::fmt::Result'>Result</a></code></h4>
<div class='docblock'><p>Formats the value using the given formatter.</p>
</div></div></section>
    <section id='search' class="content hidden"></section>

    <section class="footer"></section>

    <aside id="help" class="hidden">
        <div>
            <h1 class="hidden">Help</h1>

            <div class="shortcuts">
                <h2>Keyboard Shortcuts</h2>

                <dl>
                    <dt>?</dt>
                    <dd>Show this help dialog</dd>
                    <dt>S</dt>
                    <dd>Focus the search field</dd>
                    <dt>&larrb;</dt>
                    <dd>Move up in search results</dd>
                    <dt>&rarrb;</dt>
                    <dd>Move down in search results</dd>
                    <dt>&#9166;</dt>
                    <dd>Go to active search result</dd>
                    <dt>+</dt>
                    <dd>Collapse/expand all sections</dd>
                </dl>
            </div>

            <div class="infos">
                <h2>Search Tricks</h2>

                <p>
                    Prefix searches with a type followed by a colon (e.g.
                    <code>fn:</code>) to restrict the search to a given type.
                </p>

                <p>
                    Accepted types are: <code>fn</code>, <code>mod</code>,
                    <code>struct</code>, <code>enum</code>,
                    <code>trait</code>, <code>type</code>, <code>macro</code>,
                    and <code>const</code>.
                </p>

                <p>
                    Search functions by type signature (e.g.
                    <code>vec -> usize</code> or <code>* -> vec</code>)
                </p>
            </div>
        </div>
    </aside>

    

    <script>
        window.rootPath = "../";
        window.currentCrate = "google_genomics1";
        window.playgroundUrl = "";
    </script>
    <script src="../jquery.js"></script>
    <script src="../main.js"></script>
    
    <script defer src="../search-index.js"></script>
</body>
</html>